### PR TITLE
add messages to ok call in createDeployment

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
Locally I forgot to add `--environment=local` and I kept getting this unhelpful error:
Screenshot 2023-08-11 at 22 28 17

It took me quite a while to figure out what the problem is since debugging and stepping through the code for some reason seems not to work as it should (and the error appeared to be thrown in random places not where the assertion was actually getting made)

I've added an error message to the `ok` call that was actually erroring so that next time I can at least look for the error and understand right away the issue:
Screenshot 2023-08-11 at 22 31 26

Since I was already at it I added messages to all the `ok` calls as those could really help avoiding some pain in the future (and I don't think there is any benefit in not adding proper error messages right?)

### CODE REVIEW

File: scripts/src/createDeployment.ts [hunk 1]
```diff
@@ -242,13 +242,16 @@ export const createDeployment = async ({
 		return { url };
 	} else {
 		const host = HOSTS[environment];
-		ok(host);
+		ok(host, "The cloudflare host couldn't be determined");
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:245]:
```suggestion
		ok(host, "The Cloudflare host couldn't be determined");
```

File: scripts/src/createDeployment.ts [hunk 2]
```diff
@@ -242,13 +242,16 @@ export const createDeployment = async ({
...
 		const pagesProject = PAGES_PROJECTS[environment][trigger];
-		ok(pagesProject);
+		ok(pagesProject, "The pages project couldn't be determined");
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:247]:
```suggestion
		ok(pagesProject, "The Pages project couldn't be determined");
```

File: scripts/src/createDeployment.ts [hunk 3]
```diff
@@ -242,13 +242,16 @@ export const createDeployment = async ({
...
 
 		let id: string;
 		if (trigger !== Trigger.DirectUpload) {
-			ok(pagesProject.GIT_REPO);
+			ok(
+				pagesProject.GIT_REPO,
+				"The pages project github repo couldn't be determined"
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:253]:
```suggestion
				"The Pages project's Git repo couldn't be determined"
```

File: scripts/src/createDeployment.ts [hunk 4]
```diff
@@ -270,6 +273,7 @@ export const createDeployment = async ({
 				},
 			});
 
+			console.log(`\x1b[31m Aaaaaaaaaaa \x1b[0m`);
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:276]:
```suggestion
```

File: scripts/src/createDeployment.ts [hunk 5]
```diff
@@ -322,7 +326,7 @@ export const createDeployment = async ({
 							},
 						}
 					);
-					ok(deployHookDeletionResponse.ok);
+					ok(deployHookDeletionResponse.ok, "Deploy hook deletion failed");
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:329]:
```suggestion
					ok(deployHookDeletionResponse.ok, "Deploy Hook deletion failed");
```

File: scripts/src/createDeployment.ts [hunk 6]
```diff
@@ -359,7 +363,7 @@ export const createDeployment = async ({
 					deployHookResponseText
 				) as DeployHookResponse;
 
-				ok(result.id);
+				ok(result.id, "Missing id in deploy hook response");
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:366]:
```suggestion
				ok(result.id, "Missing id in Deploy Hook response");
```

File: scripts/src/createDeployment.ts [hunk 7]
```diff
@@ -492,7 +496,10 @@ export const createDeployment = async ({
 		fixtureConfig: FixtureConfig;
 		featuresConfig: FeaturesConfig;
 	}) {
-		ok([Environment.Production, Environment.Staging].includes(environment));
+		ok(
+			[Environment.Production, Environment.Staging].includes(environment),
+			"Wrong environment specified"
```
Comment (GregBrimble) [inline scripts/src/createDeployment.ts:501]:
```suggestion
			"Invalid environment specified"
```

**Review by GregBrimble — APPROVED** [review]

File: scripts/src/createDeployment.ts (hunk 4 above)
Comment (dario-piotrowicz) [reply scripts/src/createDeployment.ts:276]: 🤦

**Pull request merged**